### DISCUSSION TITLE
Deployed nextjs to vercel but site with disbaled JS doesn't show up

### DISCUSSION THREAD

**Discussion by herol3oy** [vercel/next.js#20225] [Help]
I have deployed a NextJS blog to Vercel but whenever I bring the website and then disable the browser JavaScript, the website doesn't show up/work properly. I thought with NextJS everything will be build and send to user so it shows everything even with disabled JavaScript! (I'm using css-in-js and sanity)

[Website](https://exss.ir/akKIt)
[Repo](https://exss.ir/R3AfJ)

**Comment by timneutkens** [comment]
Looking at the html you have a `opacity: 0` set on all elements which means they're not visible.

**Reply by herol3oy** [reply]
It has been set by framer-motion for some transition effects. But the point is that css-in-js didn't build before sending to client/user! Did I missed something or there is a conflict with my Styled-component?

**Reply by timneutkens** [reply]
The styling is there, you just don't see the items because `opacity: 0` is set. This is unrelated to how Next.js works / deploying on Vercel given that you're rendering something using Javascript (framer motion uses JS) so it's expected it does not work without JS.
